File: app/src/at-tamle/java/asiantech/internship/winter/ex17/ascending.java
package asiantech.internship.winter.ex17;

import java.util.Scanner;
//day tang dan
public class ascending {
    public static int process(){
        Scanner input= new Scanner(System.in);
        boolean check= false;
        int n=0;
        while(!check){
            System.out.print(" ");
            try{
                n= input.nextInt();
                check= true;
            }catch(Exception e){
                System.out.println(" Ban phai nhap so! hay nhap lai");
                input.nextLine();
            }
        }
        return (n);
    }
    private static float innumber(){
        Scanner input= new Scanner(System.in);
        boolean check= false;
        float n=0;
        while(!check){
            System.out.print(" ");
            try{
                n= input.nextInt();
                check= true;
            }catch(Exception e){
                System.out.println(" Ban phai nhap so! hay nhap lai");
                input.nextLine();
            }
        }
        return (n);
    }
    private static int minnumber(float a[], int n){
        float min= a[0];
        int key= 0;
        for(int j=0 ; j<n ; j++){
            if(min>a[j]){
                min= a[j];
                key= j;
            }
        }
        return (key);
    }
    public static float maxnumber(float a[], int n){
        float max= a[0];
        for(int j=0 ; j<n ; j++){
            if(max<a[j]) max= a[j];
        }
        return (max);
    }
    public static void main(String[] args) {
        int n,i;
        System.out.println("Nhap n= ");
        n= process();
        float[] array= new float[n];
        for(i=0 ; i<n ; i++){
            System.out.println("Nhap phan tu thu " +(i+1)+" ");
            array[i]= innumber();
        }
        i =0;
        System.out.println("Sap xep theo thu tu tang dan");
        while(i<n){
            System.out.println(" "+array[minnumber(array, n)]);
            array[minnumber(array, n)]= maxnumber(array, n);
            i++;
        }
    }
}

File: app/src/at-tamle/java/asiantech/internship/summer/Kotlin/Ascending.kt
package asiantech.internship.summer.Kotlin

import java.util.Scanner

object Ascending {
    fun process(): Int {
        val input = Scanner(System.`in`)
        var n = 0
        n = input.nextInt()
        return n
    }

    private fun number(): Float {
        val input = Scanner(System.`in`)
        var check = false
        var n = 0f
        while (!check) {
            print(" ")
            try {
                n = input.nextInt().toFloat()
                check = true
            } catch (e: Exception) {
                println(" Ban phai nhap so! hay nhap lai")
                input.nextLine()
            }

        }
        return n
    }

    private fun minNumber(a: FloatArray, n: Int): Int {
        var min = a[0]
        var key = 0
        for (j in 0 until n) {
            if (min > a[j]) {
                min = a[j]
                key = j
            }
        }
        return key
    }

    private fun maxNumber(a: FloatArray, n: Int): Float {
        var max = a[0]
        for (j in 0 until n) {
            if (max < a[j]) max = a[j]
        }
        return max
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val n: Int
        var i: Int
        println("Nhap n= ")
        n = process()
        val array = FloatArray(n)
        i = 0
        while (i < n) {
            println("Nhap phan tu thu " + (i + 1) + " ")
            array[i] = number()
            i++
        }
        i = 0
        println("Sap xep theo thu tu tang dan")
        while (i < n) {
            println(" " + array[minNumber(array, n)])
            array[minNumber(array, n)] = maxNumber(array, n)
            i++
        }
    }
}
